File: ruby/grains/grains.rb
class Grains
  def self.square(n)
    if n < 1 || n > 64
      raise ArgumentError
    end

    return 2 ** (n-1)
  end

  def self.total
    (0..(64-1)).map { |x| 2 ** x }.reduce(:+)
  end
end

File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64)
            throw new ArgumentOutOfRangeException();

        return (ulong)Math.Pow(2, n - 1);
    }

    public static ulong Total()
    {
        // The answer is 2^64 - 1, which is the highest value that can
        // be stored in a ulong/UInt64.
        return UInt64.MaxValue;
    }
}
